Counterexamples/TopologistsSineCurve.lean

```diff
@@ -0,0 +1,211 @@
+      refine tendsto_nhds_unique ?_ hf_lim.2
+      convert this.tendsto.comp hf_lim.1 with n
+      obtain ⟨y, hy⟩ := hf_mem n
+      simp [← hy.2]
+  · -- Show that every `p ∈ T` is the limit of a sequence in `S`.
+    rintro (hz | ⟨z, hz⟩)
+    · -- Point is in `S`: use constant sequence
+      exact ⟨_, fun _ ↦ hz, tendsto_const_nhds, tendsto_const_nhds⟩
+    · -- Point is in `Z`: use sequence from `xSeq`
+      simp only [Prod.mk.injEq] at hz
+      rcases hz with ⟨hz, ⟨rfl, rfl⟩⟩
+      refine ⟨fun n ↦ (xSeq z n, z), fun n ↦ ⟨_, xSeq_pos z n, ?_⟩, xSeq_tendsto z,
+        tendsto_const_nhds⟩
+      simpa using sin_inv_xSeq hz n
+
+lemma isClosed_T : IsClosed T := by simpa only [← closure_S] using isClosed_closure
+
+/-!
+## `T` is connected
+-/
+/-- `T` is connected, being the closure of the set `S` (which is obviously connected since it
+is a continuous image of the positive real line). -/
+theorem isConnected_T : IsConnected T := by
+  rw [← closure_S]
+  refine (isConnected_Ioi.image _ <| continuousOn_id.prodMk ?_).closure
+  exact continuous_sin.comp_continuousOn <| continuousOn_inv₀.mono fun _ hx ↦ hx.ne'
+
+/-!
+## `T` is not path-connected
+-/
+
+lemma zero_mem_T : (0, 0) ∈ T := by
+  refine .inr ⟨0, ⟨?_, ?_⟩, rfl⟩ <;>
+  linarith
+
+/-- A point in the `body` of the topologist's sine curve. -/
+noncomputable def w : ℝ × ℝ := (1, sin 1⁻¹)
+
+lemma w_mem_T : w ∈ T := .inl ⟨1, ⟨zero_lt_one' ℝ, rfl⟩⟩
+
+private lemma norm_ge_abs_snd {a b : ℝ} : |b| ≤ ‖(a, b)‖ := by simp
+
+private lemma exists_unitInterval_gt {t₀ : unitInterval} (ht₀ : t₀ < 1) {δ : ℝ} (hδ : 0 < δ) :
+    ∃ t₁, t₀ < t₁ ∧ dist t₀ t₁ < δ := by
+  let s₀ := (t₀ : ℝ) -- t₀ is in unitInterval
+  let s₁ := min (s₀ + δ/2) 1
+  have h_s₀_delta_pos : 0 ≤ s₀ + δ/2 := add_nonneg t₀.2.1 (by positivity)
+  have hs₁ : 0 ≤ s₁ := le_min h_s₀_delta_pos zero_le_one
+  have hs₁': s₁ ≤ 1 := min_le_right ..
+  refine ⟨⟨s₁, hs₁, hs₁'⟩, lt_min ((lt_add_iff_pos_right _).mpr (half_pos hδ)) ht₀, ?_⟩
+  have h_le : s₁ ≤ s₀ + δ/2 := min_le_left _ _
+  have h_ge : s₀ ≤ s₁ := le_min (by linarith) t₀.2.2
+  rw [Subtype.dist_eq, dist_comm, dist_eq, abs_of_nonneg (by linarith)]
+  linarith
+
+private lemma mem_S_of_x_pos {p : ℝ × ℝ} (hx : 0 < p.1) (hT : p ∈ T) : p.2 = sin (p.1)⁻¹ := by
+  obtain ⟨x, -, hx⟩ : p ∈ S := by
+    cases hT with
+    | inl hT => trivial
+    | inr hZ => obtain ⟨y, ⟨-, rfl⟩⟩ := hZ; exact (lt_irrefl _ hx).elim
+  simp [← hx]
+
+/-- For any `0 < a` and any `y ∈ Icc (-1) 1`, we can find `x ∈ Ioc a 0` with `sin x⁻¹ = y`. -/
+lemma exists_mem_Ioc_of_y {y : ℝ} (hy : y ∈ Icc (-1) 1) {a : ℝ} (ha : 0 < a) :
+    ∃ x ∈ Ioc 0 a, sin x⁻¹ = y := by
+  obtain ⟨N, h_dist⟩ := (Metric.tendsto_nhds.mp (xSeq_tendsto y) (a/2) (by positivity)).exists
+  refine ⟨xSeq y N, ⟨xSeq_pos y N, ?_⟩, sin_inv_xSeq hy _⟩
+  rw [dist_eq, sub_zero, abs_of_pos (xSeq_pos _ N)] at h_dist
+  linarith
+
+/-- The set `T` is not path-connected. -/
+theorem T_is_not_path_conn : ¬ IsPathConnected T := by
```

Comment:
I suggest `not_isPathConnected_T` (is this mentioned in the module comment? If so, please update that too)